### PR TITLE
FirstDraftTutorial30

### PR DESCRIPTION
**Pull request by davidRFB**
# Pull Request Template

## Description
This is the first draft for the tutorial of a DL and ML model to predict melting temperature using information obtained from ProThermDB. 
I am having troubles using deepchem keras model and also to reshape the date in  a NumpyDataset to include a fix length vector in a supportMachineRegressor. 

I have a piece of code that also download the PDBs from the RSCB but I do not include it becase I am  now working in sequence features. 

I addition to DeepChem I used propy  https://pypi.org/project/propy3/ 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by davidRFB** [comment]
Hi @ncfrey ! Here is the PR of the tutorial that I mentioned to you. Please let me know any suggestion or changes.

**Comment by ncfrey** [comment]
Thanks for putting this up @davidRFB! A few general comments
* To set up the DeepChem environment in Colab, I've moved to `condacolab`, which is easy to set up like this
```
!pip install -q condacolab
import condacolab
condacolab.install_mambaforge()  # or condacolab.install_miniconda()
```
* Can you add some documentation at the top that includes a more thorough description of what the user will learn throughout the tutorial? 
* The tutorial has a lot of headings, which makes the Table of Contents a bit difficult to navigate. Can you consolidate some of these headings and add regular text cells to explain what is going on in each section?
* Are there any functions or code blocks that might be good to add to DeepChem? Each DeepChem tutorial is generally pretty short, and consolidating utility code into DeepChem can be a nice way to add functionality and clean things up.
* You can check out some of the other tutorials to add the usual formatting, like the "Join the community" text at the bottom.

**Comment by davidRFB** [comment]
@ncfrey  Thank you very much for your comments.  I changed the headings and everything looks more organized. 
Regarding this question. 
>     * Are there any functions or code blocks that might be good to add to DeepChem? Each DeepChem tutorial is generally pretty short, and consolidating utility code into DeepChem can be a nice way to add functionality and clean things up.

I guess that some functions such as the download of sequences by accession number and, the mutation of strings could be added in some way to DeepChem, maybe if the Uniprot database is included https://github.com/deepchem/deepchem/issues/2330#issue-771134416 ( In the future I would like to contribute in that). Regarding the tutorial I thinkl is too long. So if I remove the  Data Extraction section, the tutorial can start from the small dataset that already contains the sequences and the temperatures. Therefore, the tutorial will only include the Deep Learning and Machine Learning models using protein sequences.

File: examples/tutorials/.virtual_documents/Tutorial30_Protein_DeepLearning.ipynb.py [hunk 1]
```diff
@@ -0,0 +1,345 @@
+get_ipython().getoutput("curl -Lo conda_installer.py https://raw.githubusercontent.com/deepchem/deepchem/master/scripts/colab_install.py")
```
Comment (rbharath) [inline examples/tutorials/.virtual_documents/Tutorial30_Protein_DeepLearning.ipynb.py:1]: This file should be removed I think

File: examples/tutorials/.ipynb_checkpoints/Tutorial30_Protein_DeepLearning-checkpoint.ipynb [hunk 2]
```diff
@@ -0,0 +1,1489 @@
+{
```
Comment (rbharath) [inline examples/tutorials/.ipynb_checkpoints/Tutorial30_Protein_DeepLearning-checkpoint.ipynb:1]: This file should be removed I think

**Comment by davidRFB** [comment]
Hi @rbharath . Here is the modified version of the Tutorial of protein machine learning. No prothermDB data. I am having troubles using the KerasModel of deepchem, maybe you give me your opinion about it.  It is in section : DeepChem Keras Model.

**Review by rbharath — COMMENTED** [review]
Sorry for the very late review! Finally getting caught up on my backlog of DeepChem work. Here are some comments:

- Can you remove the checkpoint files from this PR?
- Typo: "In this tutorial"
- There are a lot of comments in here that need to be cleaned up still
- For processing code, make sure to explain why you're running a particular function with some text that explains the reasoning behind what's happening
- Typo: "PDB Download"
- There are large print displays that should be cleaned up so the tutorial reads nicer
- Hmm, the error here on `KerasModel` is strange. I don't know offhand what's causing it, but once the rest of the tutorial is in good shape will do a run through and try to fix it
-

**Comment by davidRFB** [comment]
@rbharath  No worries at all, I already improve some comments and add more explanation to others cells, and delete the checkpoint files. So if you see anything else related, please let me know. Thank you very much.

**Comment by rbharath** [comment]
I think the comments from my last review still need to be addressed. Once those are hit will do a second thorough review

**Comment by davidRFB** [comment]
Hi @rbharath  ! Here are more explanation and comments in intermediate steps along the tutorial. Please let me know any suggestion or change. !

**Review by rbharath — COMMENTED** [review]
Looks better! The big open issue is fixing the DeepChem keras model usage. Can you try to see how that can be made to work?

@alat-rights You might be interested in reading this tutorial as well since it has some overlap with things you've been thinking of

**Comment by davidRFB** [comment]
Hi ! I think that I solved the problem with the keras model. I had a mistake in the loss function. At this moment, the both models run normally. I would like to know any other suggestions.

**Review by rbharath — APPROVED** [review]
Looks good! I'm going to go ahead and merge this in. Congrats on the merged PR :)
